### PR TITLE
Validation  Asset forms 

### PR DESCRIPTION
**Pull request by clairecfu**
- [x] mac address validation on front end 
- [x] mac address validation on back end 
- [x] convert empty strings to undefined for asset name and hostname

### CODE REVIEW

File: rackcity/models/network_port.py [hunk 1]
```diff
@@ -1,14 +1,17 @@
 from django.db import models
 from .asset import Asset
-from django.core.exceptions import ObjectDoesNotExist
-
```
Comment (clairecfu) [inline rackcity/models/network_port.py]: linter was yelling at me

File: rackcity/models/network_port.py [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 
+def format_mac_address(mac_address):
+    # all lower case
+    mac_address = mac_address.lower()
+    # if contains - as a delimiter, convert to :
+    mac_address = mac_address.replace("-", ":")
+    # if contains no delimiters:
+    mac_address = '-'.join(a+b for a, b in zip(mac_address[::2], mac_address[1::2]))
+    return mac_address
+
```
Comment (clairecfu) [inline rackcity/models/network_port.py]: actual logic for formatting mac address

File: rackcity/models/network_port.py [hunk 3]
```diff
@@ -20,33 +23,28 @@ class NetworkPort(models.Model):
...
+                + "' is already connected to port '"
+                + destination_port.connected_port.asset.hostname
+                + ":"
+                + destination_port.connected_port.port_name
+                + "'. "
             )
         if self.connected_port:
             self.delete_network_connection()
+        if self.mac_address is not None:
+            self.mac_address = format_mac_address(self.mac_address)
```
Comment (clairecfu) [inline rackcity/models/network_port.py]: this is where I call the function and update the value

File: src/forms/formUtils.tsx [hunk 4]
```diff
@@ -17,6 +22,21 @@ export function escapeRegExpChars(text: string) {
...
+    return true;
+  }
+  return false;
+}
+
+export const macAddressInfo = (
+  <p>
+    6-byte hexadecimal string. <em>Valid delimiters: none, :, - </em>
+  </p>
+);
```
Comment (clairecfu) [inline src/forms/formUtils.tsx]: regex for front end mac address validation. do we want other delimiters other than "-" or ":"?

File: rackcity/models/network_port.py [hunk 5]
```diff
@@ -65,6 +74,12 @@ def delete_network_connection(self):
...
-                name='unique network port names on assets'),
+                fields=["asset", "port_name"],
+                name="unique network port names on assets",
+            ),
         ]
+
+    def save(self, *args, **kwargs):
+        if self.mac_address is not None:
+            self.mac_address = format_mac_address(self.mac_address)
+        super(NetworkPort, self).save(*args, **kwargs)
```
Comment (juliasaveliff) [inline rackcity/models/network_port.py]: I approve if you confirm that the front end will handle character validation (ie everything between delimiters is hex 0-9, a-f)

Comment (clairecfu) [reply rackcity/models/network_port.py]: yes, the backend should only every recieve mac addresses that are 6 byte hex strings with no delimiters, :, or - delimiters

**Pull request merged**